### PR TITLE
009 login system

### PR DESCRIPTION
**Pull request by ashleighB31**

### CODE REVIEW

File: views/homepage.html [hunk 1]
```diff
@@ -10,6 +10,21 @@
...
+        <link href="https://fonts.cdnfonts.com/css/galano-grotesque-alt" rel="stylesheet">
+        <script src="https://kit.fontawesome.com/2ea65b5d39.js" crossorigin="anonymous"></script>
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+        
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
+        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+        
+    </head>
```
Comment (anne-moore) [inline views/homepage.html:27]: There are 2 tags to end the head element

File: views/jobRolesList.html [hunk 2]
```diff
@@ -11,6 +12,20 @@
...
+
+        <link href="https://fonts.cdnfonts.com/css/galano-grotesque-alt" rel="stylesheet">
+        <script src="https://kit.fontawesome.com/2ea65b5d39.js" crossorigin="anonymous"></script>
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+        
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
+        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+    </head>
```
Comment (anne-moore) [inline views/jobRolesList.html]: 2 tags to end head element

**Review by anne-moore — APPROVED** [review]

File: src/app.ts [hunk 3]
```diff
@@ -36,5 +40,7 @@ app.listen(3000, () => {
     console.log('Server started on port 3000');
 });
 
-app.get('/homepage' , getHomePage)
 app.get('/jobRoles', getAllJobRoles);
+app.get('/loginForm', getLoginForm);
+app.post('/loginForm', postLoginForm);
```
Comment (shaunganley) [inline src/app.ts]: why not put these in new auth routes file?

File: src/controllers/AuthController.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        res.render('loginForm.html', { ...req.body, errormessage: res.locals.errormessage });
+    }
+}
+
+export const logout = async (req: express.Request, res: express.Response): Promise<void> => {
+    req.session.destroy((err) => {
+        if (err) {
+            res.status(500).send('Failed to log out');
+        } else {
+            res.status(200).send('Logged out successfully');
```
Comment (shaunganley) [inline src/controllers/AuthController.ts]: redirect to login

File: src/services/AuthService.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { AxiosResponse } from "axios";
+import { LoginRequest } from "../models/LoginRequest";
+import { requestInstance } from "../models";
+
+export const getToken = async (loginRequest: LoginRequest): Promise<string> => {
+    try {
+        const response: AxiosResponse = await requestInstance.post("http://localhost:8080/api/auth/login", loginRequest);
```
Comment (shaunganley) [inline src/services/AuthService.ts]: don't hardcode base url

File: test/unit/controller/AuthControllerTest.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ 
+const mockToken = "mockedtoken";
+ 
+ 
+describe('AuthController', function () {
+    afterEach(() => {
+        sinon.restore();
+    });
+ 
+    describe('getLoginForm', function () {
```
Comment (shaunganley) [inline test/unit/controller/AuthControllerTest.ts:16]: add tests for logout method

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/controllers/AuthController.ts (hunk 4 above)
Comment (ashleighB31) [reply src/controllers/AuthController.ts]: Screenshot 2024-07-31 at 13 57 42

We are currently using the html to redirect to the login page, do you still want this done in the controller?

File: test/unit/controller/AuthControllerTest.ts (hunk 6 above)
Comment (ashleighB31) [reply test/unit/controller/AuthControllerTest.ts:16]: tests for current logout function done

**Review by shaunganley — APPROVED** [review]

**Pull request merged**